src/config/environmentConfig.ts

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+/**
+ * Validates and sets the environment configuration based on the provided schema.
+ * If the environment config is invalid, an error is thrown.
+ *
+ * @throws {Error} If the environment config is invalid
+ *
+ * @returns {EnvironmentConfig} The validated environment configuration
+ */
+
+import * as joi from '@hapi/joi';
+import { EnvironmentConfig } from '../models/environmentConfig.interface';
+
+const envs = ['local', 'development', 'qa', 'production', 'test'];
+
+if ([envs[0], envs[4]].includes(process.env.NODE_ENV!)) {
+  /* eslint-disable  @typescript-eslint/no-var-requires */
+  require('dotenv').config();
+}
+
+const schema = joi.object().keys({
+  port: joi.string().default('3000'),
+  env: joi
+    .string()
+    .valid(...envs)
+    .default(envs[0]),
+  appInsightsKey: joi.string().allow('').default(''),
+  azureKeyVaultURL: joi.string().allow('').default(''),
+  geoNetworkSearchAPI: joi.string().allow('').default(''),
+});
```

Comment:
move the schema to a different folder & file. so that we can unit test them.

Reply:
Yes, I moved it to another folder and the test cases have been implemented.